### PR TITLE
Adding utility function for calculating model size in bytes

### PR DESCRIPTION
**Pull request by calebrob6**
## What does this PR do?

Adding a utility function for computing the size of a model per discussion in #8343

### Does your PR introduce any breaking changes ? If yes, please list them.

No breaking changes.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)
- [x] Did you list all the breaking changes introduced by this pull request?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/model_helpers.py [hunk 1]
```diff
@@ -69,3 +71,28 @@ def is_overridden(
     parent_code = parent_attr.__code__
 
     return instance_code != parent_code
+
+
+def get_model_size(model: nn.Module) -> int:
```
Comment (ananthsub) [inline pytorch_lightning/utilities/model_helpers.py:76]: 1. this could be under https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/utilities/memory.py
2. `get_model_size_bytes` is explicit that the return will be in bytes

File: tests/utilities/test_model_helpers.py [hunk 2]
```diff
@@ -92,3 +94,28 @@ def on_fit_start(self):
...
+
+def test_get_model_size():
+    model = BoringModel()
+
+    size_bytes = get_model_size(model)
+
+    # The BoringModel has a fully connected layer of size 32x2 with a bias resulting in
+    # 67 weights. Each weight is a float32 -- 4 bytes, therefore we expect a size of
+    # 264.
+    assert size_bytes == 264
```
Comment (ananthsub) [inline tests/utilities/test_model_helpers.py:107]: prefer defining a module to test directly here vs importing it. if someone changes the BoringModel changes for any reason, this test will fail

File: tests/utilities/test_model_helpers.py [hunk 3]
```diff
@@ -92,3 +94,28 @@ def on_fit_start(self):
...
+
+
+def test_get_sparse_model_size():
+
+    class BoringSparseModel(BoringModel):
+
+        def __init__(self):
+            super().__init__()
+            self.layer = nn.Parameter(torch.ones(32).to_sparse())
```
Comment (ananthsub) [inline tests/utilities/test_model_helpers.py:116]: same as above

File: pytorch_lightning/utilities/model_helpers.py (hunk 1 above)
Comment (tchaton) [reply pytorch_lightning/utilities/model_helpers.py:76]: Should utility be part of the training_type_plugin as shared tensors won't provide the right value ?

File: tests/utilities/test_model_helpers.py (hunk 2 above)
Comment (tchaton) [reply tests/utilities/test_model_helpers.py:107]: The BoringModel shouldn't change :) If it does, lot of test will fail :)

**Comment by awaelchli** [comment]
@calebrob6 I merged master and added a changelog entry. 
Ready for review? 
This will then unblock #8495

**Comment by calebrob6** [comment]
Hey @awaelchli, I just requested a review, although I have not changed anything about the code since the above comments.

I think (from #8495) that this code should be merged as-is as an utility function, the model_size() property will be deprecated, then I will implement a new version of this utility function that deals with sparse tensors.

**Comment by awaelchli** [comment]
Yesterday #8495 was labelled as "blocked by 8500". 

@roshikouhai @ananthsub @calebrob6 can you confirm in which order these two PRs should be reviewed and merged?

**Review by Borda — APPROVED** [review]

**Comment by Borda** [comment]
> can you confirm in which order these two PRs should be reviewed and merged?

I think this shall be merged before deprecating size attribute

File: tests/utilities/test_model_helpers.py [hunk 4]
```diff
@@ -88,3 +90,26 @@ def on_fit_start(self):
...
+
+def test_get_sparse_model_size():
+    class BoringSparseModel(BoringModel):
+        def __init__(self):
+            super().__init__()
+            self.layer = nn.Parameter(torch.ones(32).to_sparse())
+
+    model = BoringSparseModel()
+
+    with pytest.raises(NotImplementedError):
```
Comment (awaelchli) [inline tests/utilities/test_model_helpers.py:114]:
```suggestion
    with pytest.raises(NotImplementedError, match="sparse tensors is not implemented"):
```
for the sanity ^^

**Review by awaelchli — APPROVED** [review]

**Comment by tchaton** [comment]
Dear @calebrob6,

Thanks for putting the effort with this PR. The Lightning Team decided to go with the ByteCounter approach for v1.5. in this PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/10123

Closing this PR.

Best,
T.C
